internal/deviations/deviations.go

```diff
@@ -95,16 +95,24 @@ func lookupDeviations(dut *ondatra.DUTDevice) (*mpb.Metadata_PlatformExceptions,
 	return matchedPlatformException, nil
 }
 
-func mustLookupDeviations(dut *ondatra.DUTDevice) *mpb.Metadata_PlatformExceptions {
-	platformExceptions, err := lookupDeviations(dut)
+func mustLookupDeviations(dvc *ondatra.Device) *mpb.Metadata_PlatformExceptions {
```

Comment:
heh, mustLookupDeviations doesn't actually return Deviations

It looks like you can move more or less all the logic shared by lookupDUTDeviations and lookupATEDeviations into here, and change this to return Deviations. Then the implementations of lookupDUTDeviations  and lookupATEDeviations become:

`return mustLookupDeviations(dut.Device)` and `return mustLookupDeviations(ate.Device)`, respectively